diagnostics/albedofb/albedofb_calcs.py

```diff
@@ -177,29 +197,28 @@ def process_data(kernel_file, sensitivity_file):
     tas = xr.open_dataset(TAS_input_file)
     tas = tas[TAS_var]
     fx = xr.open_dataset(area_input_file)
-    fx = fx['areacella']
+    fx = fx[area_var]
     Tglob = globalmean(tas, fx)
     Tglobtrend = Tglob.groupby('time.month').apply(linear_trend)
 
-    ds = xr.open_dataset(FSDS_input_file)
-    us = xr.open_dataset(FSUS_input_file)  
-    # must get rid of cftime since lack of leap day messes up divide below
-    ds=ds['rsds'].assign_coords(month=('time',ds.time.dt.month)).swap_dims({'time':'month'}).drop('time')
-    us=us['rsus'].assign_coords(month=('time',us.time.dt.month)).swap_dims({'time':'month'}).drop('time')
-    albedo = us/ds
-    stacked=albedo.stack(allpoints=['lat','lon']) # reduce to 2D array before computing trend
-    albedo_trend=stacked.groupby(('month')).apply(linear_trend) # loses coord to unstack somewhere here
-    tmp=stacked.isel(month=slice(0,12))           # use this to put back stack coord
-    tmp[:,:]=albedo_trend
+    fsds = xr.open_dataset(FSDS_input_file)
+    fsus = xr.open_dataset(FSUS_input_file)
+
+    ds = fsds['rsds']
+    us = fsus['rsus']
+    albedo = us / ds
+    stacked = albedo.stack(allpoints=['lat', 'lon'])  # reduce to 2D array before computing trend
+    albedo_trend = stacked.groupby('time.month').apply(linear_trend)
+    tmp = stacked.isel(time=slice(0, 12))  # use this to put back stack coord
+    tmp[:, :] = albedo_trend
     albedo_trend = tmp.unstack('allpoints')
-    albedo_trend=albedo_trend.fillna(0)
-    albedo_per_Tglobtrend=albedo_trend/Tglobtrend
+    albedo_trend = albedo_trend.fillna(0)
+    albedo_per_Tglobtrend = albedo_trend / Tglobtrend
     albedo_per_Tglobtrend.name = 'sensitivity'
     albedo_per_Tglobtrend.to_netcdf(sensitivity_file)
```

Comment:
This is the only piece where I am getting a little lost.  I am not sure what the old code is trying to do w.r.t. leap years. The rest of the mods look innocuous. Is there a test case that can be run on both versions to check the results?

Reply:
@jkrasting The original problem wasn't clear to me, either (maybe this was left over from an earlier iteration of the code), and the new mods work with the original dataset. I'll run a calculation check for albedo_trend with the new and old code, and report back.

Reply:
@jkrasting My mods changed the answers, but I was able to revert them and replace the ds.time.dt.month with ds.time.dt.month.data in assign_coords to resolve the original issue.